app/summary/page.tsx

```diff
@@ -0,0 +1,143 @@
+"use client";
+
+import untypedSubjectSummaries from "@/app/generated/subject-summaries.json";
+import { SummaryRow } from "@/app/types";
+import { useState, useRef, useEffect } from "react";
+
+const SUMMARY_DISPLAY_COUNT = 100;
+const subjectSummaries = untypedSubjectSummaries as unknown as SummaryRow[];
+const tableHeaders = [
+    {
+        key: "enrollmentTerm",
+        label: "Term"
+    },
+    {
+        key: "subjectArea",
+        label: "Subject"
+    },
+    {
+        key: "catalogNumber",
+        label: "Course Number"
+    },
+    {
+        key: "courseTitle",
+        label: "Course Name"
+    },
+    {
+        key: "instructorName",
+        label: "Professor"
+    },
+    {
+        key: "gradeAverage",
+        label: "Average Grade"
+    },
+    {
+        key: "gradeSpread",
+        label: "Grade Spread"
+    },
+    {
+        key: "gradedTotal",
+        label: "Total Graded"
+    },
+    {
+        key: "enrollmentTotal",
+        label: "Total Enrolled"
+    }
+]
+
+const Summary = () => {
+    const [cutoff, setCutoff] = useState(SUMMARY_DISPLAY_COUNT);
```

Comment:
instead of a fixed constant, let's make the initial value `cutoff` a prop of `Summary` with a default value of `100` so then the initial display count can be customized in case we want to use this component in other places with different layout/space constraints (e.g. we might want to show the top 5 classes with highest average on the main page)